newIDE/app/src/ObjectsList/index.js

```diff
@@ -323,18 +538,25 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
       })
     );
 
+    const scrollToItem = React.useCallback((itemId: string) => {
+      if (treeViewRef.current) {
+        treeViewRef.current.scrollToItemFromId(itemId);
+      }
+    }, []);
+
     const addObject = React.useCallback(
       (objectType: string) => {
         const defaultName = project.hasEventsBasedObject(objectType)
           ? 'New' +
             (project.getEventsBasedObject(objectType).getDefaultName() ||
               project.getEventsBasedObject(objectType).getName())
           : objectTypeToDefaultName[objectType] || 'NewObject';
-        const name = newNameGenerator(defaultName, name =>
-          projectScopedContainersAccessor
-            .get()
-            .getObjectsContainersList()
-            .hasObjectNamed(name)
+        const name = newNameGenerator(
+          defaultName,
+          name =>
+            objectsContainer.hasObjectNamed(name) ||
+            (!!globalObjectsContainer &&
+              globalObjectsContainer.hasObjectNamed(name))
```

Comment:
Is the logic I added juste before you opened the PR (using the projectScopedContainersAccessor only) better? Or do you prefer keeping this objects containers check?

Reply:
This component relies heavily on global/scene for objects.
I guess what is important is that the component doesn't ask both `projectScopedContainersAccessor` and a couple of object container because:
  - it's risky as they could not be in sync in case of a mistake in the caller
  - when the component will be reworked, the changes will be local to the component